src/pages/LinkDevicePage/hooks.ts

```diff
@@ -20,21 +21,27 @@ const getSites = (profiles?: Profile[]) => {
 };
 
 const useSearchDevices: UseSearchDevices = (siteId, q) => {
-  const sites = useQueryWithClient([QueryKey.SITES], getSitesService);
+  const sitesResponce = useQueryWithClient([QueryKey.SITES], getSitesService);
 
-  const devices = useQueryWithClient(
+  const sites = useMemo(() => {
+    return getSites(sitesResponce.data?.data?.me?.profiles);
+  }, [sitesResponce.data]);
+
+  const devicesResponce = useQueryWithClient(
     [QueryKey.DEVICES, siteId as Site["id"], q as string],
     (client) => searchDevicesService(client, { siteId: siteId as Site["id"], q: q as string }),
     { enabled: Boolean(q) && Boolean(siteId) },
   );
 
+  const devices: DeviceType[] = useMemo(() => {
+    return enhanceSearchDevices(find(sites, { id: siteId }) as Site, devicesResponce.data?.data);
```

Comment:
If I've understhood this correctly, `enhanceSearchDevices` returns `DeviceType[]` but it typed to `Site`. Is this not a type mismatch?

Reply:
You understand correctly because the assets are located on the site.